### PR TITLE
Promote

### PR DESCRIPTION
**Pull request by stdavis**
Most of these enhancements and fixes were requested by CityWorks so that West Jordan could use our locator in their product. The main issue was that they were sending most, if not all, requests as posts.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 setup(
     name="masquerade",
-    version="1.2.2",
+    version="1.4.0-0",
```
Comment (steveoh) [inline setup.py:14]: **question**: is our versioning messed up? Shouldn't this be 1.3.0?

File: src/masquerade/utils.py [hunk 2]
```diff
@@ -14,3 +19,32 @@ def cleanse_text(text):
...
 
     return text.strip().replace('"', "").replace("'", "")
+
+
+def get_request_param(request, param_name):
+    """get the parameter from the request checking both url params and form data"""
+    if param_name in request.args:
+        return request.args.get(param_name)
+
+    return request.form.get(param_name)
```
Comment (steveoh) [inline src/masquerade/utils.py:29]: **question**: Should we log what params aren't in the request or is that not useful?

File: tests/test_utils.py [hunk 3]
```diff
@@ -14,3 +16,36 @@ def test_cleanse_text_without_strings():
...
+    request.method = "GET"
+    request.args = {"outSR": 3857}
+
+    assert get_out_spatial_reference(request) == (3857, 3857)
+
+
+def test_get_out_spatial_reference_get_request_json():
+    request = MagicMock()
+    request.method = "GET"
+    request.args = {"outSR": '{"wkid": 3857}'}
```
Comment (steveoh) [inline tests/test_utils.py:32]: **question**: should we test for and use the value of `lastestWkid` if it's available?

**Review by steveoh — COMMENTED** [review]
LGTM

File: .vscode/settings.json [hunk 4]
```diff
@@ -1,9 +1,4 @@
 {
-  "python.linting.enabled": true,
```
[the remark this answers is not in the document]
Comment (stdavis) [reply .vscode/settings.json:2]: I think that at least some of these settings in agrc/python are also deprecated...

File: src/masquerade/utils.py (hunk 2 above)
Comment (stdavis) [reply src/masquerade/utils.py:29]: punted to #179

File: tests/test_utils.py (hunk 3 above)
Comment (stdavis) [reply tests/test_utils.py:32]: punted to #180

**Pull request merged**